### PR TITLE
getting-started/background: split Translate into smaller functions

### PR DESCRIPTION
**Pull request by tbpg**
I put them in the order they'll be shown in the doc. I put the cloud function (`Translate`) last so there won't be compile errors before the other helper functions are defined.

I'm open to splitting it up more or less, if you think that will be more clear.

### CODE REVIEW

File: getting-started/background/translate.go [hunk 1]
```diff
@@ -80,6 +85,40 @@ func Translate(ctx context.Context, m PubSubMessage) error {
...
 		}
 	}
+	return nil
+}
+
+// [END getting_started_background_translate_init]
+
+// [START getting_started_background_translate_string]
+
+// translateString translates text to lang, returning the translated text,
```
Comment (fhinkel) [inline getting-started/background/translate.go]: nit: I'd find this easier to read with a list of return objects.

File: getting-started/background/translate.go [hunk 2]
```diff
@@ -80,6 +85,40 @@ func Translate(ctx context.Context, m PubSubMessage) error {
...
+	}
+
+	return outs[0].Text, outs[0].Source.String(), nil
+}
+
+// [END getting_started_background_translate_string]
+
+// [START getting_started_background_translate]
+
+// Translate translates the given message and stores the result in Firestore.
```
Comment (fhinkel) [inline getting-started/background/translate.go]: Is this a convention to include the function name in the comment? I'd leave it out:
* You never risk that comment and function name diverge
* Comment is one word shorter without losing any information

(I might just be used to seeing comments like `Translate the given message` and my brain is trained to associate that with the function below)

File: getting-started/background/translate.go (hunk 1 above)
Comment (tbpg) [reply getting-started/background/translate.go]: Done (changed to a list).

File: getting-started/background/translate.go (hunk 2 above)
Comment (tbpg) [reply getting-started/background/translate.go]: Yes, it's standard for comments to start with the thing being described. https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences

**Pull request merged**